File: RESTful Samples/C#/RESTfulEngine.CSharpClient/src/DataSource.cs
/*
 * Copyright (c) 2015 by Windward Studios, Inc. All rights reserved.
 *
 * This software is the confidential and proprietary information of
 * Windward Studios ("Confidential Information").  You shall not
 * disclose such Confidential Information and shall use it only in
 * accordance with the terms of the license agreement you entered into
 * with Windward Studios, Inc.
 */

using System.Collections.Generic;
using System.Xml.Linq;

namespace RESTfulEngine.CSharpClient
{
    /// <summary>
    /// The base class for all data source providers.
    /// </summary>
    public abstract class DataSource
    {
        public IList<TemplateVariable> Variables { get; set; }

        protected XElement GetVariablesXml()
        {
            var element = new XElement("Variables");
            foreach (TemplateVariable variable in Variables)
            {
                element.Add(new XElement("Variable",
                    new XElement("Name", variable.Name),
                    new XElement("Value", variable.Value)
                    ));
            }
            return element;
        }

        internal abstract XElement GetXml(string name);
    }
}

File: RESTful Samples/RESTfulEngine.PHPClient/DataSource.php
<?php
/*
 * Copyright (c) 2015 by Windward Studios, Inc. All rights reserved.
 *
 * This software is the confidential and proprietary information of
 * Windward Studios ("Confidential Information").  You shall not
 * disclose such Confidential Information and shall use it only in
 * accordance with the terms of the license agreement you entered into
 * with Windward Studios, Inc.
 */

/** The base class for all data source.
 */
abstract class DataSource {

	public function getVariables()
	{
		return $this->state["variables"];
	}
	public function setVariables($value)
	{
		$this->state["variables"] = $value;
	}

	public abstract function getObject($name);
	
	protected function __construct()
	{
		$this->state["variables"] = NULL;
	}
	
	protected function applyVariables(&$data)
	{
		$vars = $this->getVariables();
		if (!$vars || !count($vars))
			return;

		$data["Variables"] = array();
		foreach ($vars as $var) {
			$v["Name"] = $var->getName();
			$v["Value"] = $var->getValue();
			$data["Variables"][] = $v;
		}
	}
}

?>
